packages/runtime-core/src/component.ts

```diff
@@ -561,7 +561,7 @@ export function handleSetupResult(
     }
     // setup returned bindings.
     // assuming a render function compiled from template is present.
-    instance.setupState = reactive(setupResult)
+    instance.setupState = proxyRefs(setupResult)
```

Comment:
@yyx990803 is it on purpose that `proxyRefs` doesn't check if its argument is reactive and always wraps it with a new proxy?
I very often return state that is `reactive({})` and it seems wasteful to always go 2 layers deep in proxies to access state, when just one would be perfectly fine.

I'd rather have `proxyRefs` be a default fallback when I pass a plain object.

Or going full circle to one of my very first suggestions: not wrapping at all and leaving the responsibility of picking a wrapper to `setup`. (it seems reasonable that `<script setup>` always wraps the exports in a `proxyRefs`).